src/CreatorToken.sol

```diff
@@ -99,17 +99,23 @@ contract CreatorToken is ERC721 {
   }
 
   function buy(uint256 _maxPayment) public {
-    _buy(msg.sender, _maxPayment);
+    uint256 _totalPrice = _buy(msg.sender);
+    if (_totalPrice > _maxPayment) {
```

Comment:
Instead of repeating this, this method should just curry to `buy` with arrity 2

Reply:
Curried it here: [43728e6](https://github.com/showtime-xyz/creator-tokens/pull/29/commits/43728e6415b85c2b58014ab0e5adabb84a43efa5)